GAD2.py

```diff
@@ -0,0 +1,209 @@
+                       'g': 'c', 'G': 'C', 'c': 'g', 'C': 'G'}
+    
+    def __init__(self, seq) -> None:
+        super().__init__(seq = seq)
+
+
+def check_gc(fastq_read: str, gc_params: tuple) -> bool:
+    """
+    Filters sequences in FASTQ file by GC percentage. 
+    
+    Input:
+    - fastq_read (str): FASTQ sequence read.
+    - gc_params (tuple): range for filtration, accepts upper or upper/lower border. Both borders are included.
+    
+    Output:
+    Returns boolean value is this read satisfied the criteria.
+    """
+    gc_result = SeqUtils.GC123(fastq_read)[0]
+    if gc_params[0] < gc_result < gc_params[1]:
+        return True
+
+
+def check_length(fastq_read: str, len_bound_params: tuple) -> bool:
+    """
+    Filters sequences in FASTQ file by sequence length.
+    
+    Input:
+    - fastq_read (str): FASTQ sequence read.
+    - len_bound_params (tuple): range for filtration, accepts upper or upper/lower border. Both borders are included.
+    
+    Output:
+    Returns boolean value is this read satisfied the criteria.
+    """
+    len_of_seq = len(fastq_read)
+    if len_bound_params[0] < len_of_seq < len_bound_params[1]:
+        return True
+
+
+def check_quality(fastq_quality: str, quality_params: int) -> bool:
+    """
+    Filters sequences in FASTQ file by mean quality score.
+    
+    Input:
+    - fastq_quality (str): FASTQ read quality
+    - quality_params (int): threshold value of reads quality in phred33 scale.
+    
+    Output:
+    Returns boolean value is this read satisfied the criteria.
+    """ 
+    mean_quality = sum(fastq_quality.letter_annotations["phred_quality"])/len(fastq_quality.seq)
+    if mean_quality >= quality_params:
+        return True
+
+
+def int_to_tuple(input_parameters) -> tuple:
+    """
+    Converts input parameters to tuple format.
+    If input is already a tuple, it will be return without changes.
+    If input parameter is int (only upper threshold is entered), function will return a tuple like (0, 'input').
+    
+    Input:
+    - input_parameters (tuple or int).
+    
+    Output:
+    Lower and upper threshold for functions in tuple format.
+    """
+    if isinstance(input_parameters, tuple):
+        return input_parameters
+    return (0, input_parameters)
+
+
+def run_fastq_filter(input_path: Optional[str] = None, output_filename: Optional[str] = None, gc_bounds: Union[int, tuple] = (0, 100), length_bounds: Union[int, tuple] = (0, 2**32), quality_threshold: int = 0) -> TextIO:
+    """
+    FASTQ Filtraror with BIOPYTHON utilities
+    Performs filter of input FASTQ file according to input parameters. 
+    Input will be filtered by: 
+        - GC content (gc_bounds);
+        - reads length (length_bounds);
+        - reads quality score (quality_threshold).
+        
+    Input:
+    - input_path (str): path to .fastq file; include 4 strings: 1 - read ID, 2 - sequence, 3 - comment, 4 - quality. Default - None.
```

Comment:
Кажется, описание спецификации fastq-формата уже лишнее)
```suggestion
    - input_path (str): path to .fastq file.
```